nav2_bringup/bringup/launch/nav2_bringup_launch.py

```diff
@@ -29,15 +30,32 @@ def generate_launch_description():
     launch_dir = os.path.join(bringup_dir, 'launch')
 
     # Create the launch configuration variables
+    namespace = LaunchConfiguration('namespace')
     map_yaml_file = LaunchConfiguration('map')
     use_sim_time = LaunchConfiguration('use_sim_time')
     params_file = LaunchConfiguration('params_file')
     bt_xml_file = LaunchConfiguration('bt_xml_file')
     autostart = LaunchConfiguration('autostart')
+    use_remappings = LaunchConfiguration('use_remappings')
+
+    # TODO(orduno) Remove once `PushNodeRemapping` is resolved
+    #              https://github.com/ros2/launch_ros/issues/56
+    remappings = [((namespace, '/tf'), '/tf'),
```

Comment:
Correct me if I'm reading into this wrong, but we're remapping topics like scan/cmd_vel/map for what reason? If we're running sensors they should be in their specific namespaces if the `/` wasn't added in front. I'd think it would be clearer to go through the stack and remove all the `/`. On the TF front I totally get it, but also I'm not totally understanding why there's 2 TF/TF static remaps

Reply:
In the case of `scan`, `cmd_vel` and `map`, that is correct, we could remove the leading `/` throughout the stack and have the same effect. Just thought this was good enough for now and could fix that afterward.

In the case of `tf` we have to remap both listener and broadcaster. The listener subscribes to `/tf` so I'm mapping the input `robot1/tf` to `/tf`. For the broadcaster, I remap the publisher from `/tf` to `tf` so the actual message get published (after applying the node namespace) to `robot1/tf`. And we do the same for static tfs.

Reply:
I'm totally fine with that for now, but if you file a ticket  for it, I can clean it up if you don't have the motivation to. 

On TF, that makes sense. 

> so I'm mapping the input robot1/tf to /tf

do you mean the opposite?

Reply:
I might explain it better with an example.

The Gazebo diff drive plugin publishes the odom to base_link transform to `/tf` topic (see note below). However, we need the topic in `/robot1/tf` in order to avoid collision with other robot's tf. To do this we remove the leading `/` making it a relative namespace and therefore the node applies its namespace, sending the transform on `/robot1/tf`. That's the case for any generator of transform.

The costmap instance on robot1 expects the transform to arrive on `/tf` (note below) so we remap the topic `/robot1/tf` to `/tf`. That's for any consumer of transforms.

Side node: the tf pkg has hardcoded fully-qualified topic name (with leading `/`) so any node level namespacing is not applied.

Makes sense? Or perhaps I'm missing something...

Reply:
Yes, I understand all that - I was just pointing out that I think you meant mapping `/tf` -> `robot1/tf` and not `robot1/tf` -> `/tf` as you comment mentioned. We're on the same page.

I also wonder though if the better option is to add a `prefix` variable to the TF publisher/subscriber constructors and then just instantiate all the nav2 TF things with a launchfile robot name variable. That does what we need to be done and generalizes better for other similar issues. But again, a future work thing, not hangign up this PR for that

Reply:
I agree on an upstream change to add a prefix to the tf broadcaster/listener. The broadcaster could even take the namespace off the node if not adding a separate constructor argument.

Reply:
Prefixing the transforms was actually previously supported and  [deprecated](http://wiki.ros.org/tf2/Design#tf_prefix_is_confusing_and_counterintuitive).

There is an ongoing discussion on a replacement:
https://github.com/ros/geometry2/issues/32